apps/@sparrow-desktop/src-tauri/src/main.rs

```diff
@@ -345,25 +346,45 @@ async fn make_request_v2(
     // Extract status code from response
     let response_status = response_value.status().clone();
 
-    // Extract response value from response
-    let response_text_result = decode_response_body(response_value).await;
+    //check is data binary
+
+    let is_binary = response_headers
+        .get("content-type")
+        .map(|v| v.to_str().unwrap_or("").starts_with("image/"))
+        .unwrap_or(false);
+
+    let response_body;
+
+    if is_binary {
+        //extract bytes from respose body for further conversion
+        let bytes = response_value
+            .bytes()
+            .await
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e.to_string()))?;
+
+        response_body = base64::encode(bytes); // convert bytes to base64 string effiecient for transmission / no data willl not get currupted.
+    } else {
```

Comment:
lets make a complete string here with content type

Reply:
I think we have to make complete string on response body component itself because if we need to store image in local storage it will be straight forward

Reply:
Sure

Reply:
> lets make a complete string here with content type

Got it